codecov_cli/services/staticanalysis/analyzers/python/node_wrappers.py

```diff
@@ -20,9 +56,21 @@ def do_visit(self, node):
                 "for_statement",
                 "while_statement",
             ):
-                if node.prev_named_sibling:
+                if node.type == "expression_statement" and self._is_function_docstring(
+                    node
+                ):
+                    return
+                closest_named_sibling_not_comment_that_is_in_statements = (
+                    self._get_previous_sibling_that_is_not_comment_not_func_docstring(
+                        node
+                    )
+                )
+                if closest_named_sibling_not_comment_that_is_in_statements:
```

Comment:
when would this be `None`? does this being `None` mean that `current_line_number` really doesn't have an ancestor?

Reply:
"siblings" are nodes in the same "depth" in the code. So for example, global code and function definitions are all siblings (ordered by the line they are declared).
In the same way all lines in a function (in the same scope i.e. identation) are siblings. 
So if a node is the very first line of a function it won't have a previous sibling, because it's the first node of that scope.

Yes, it means that `current_line_number` doesn't have a sure ancestor in its scope (so we will look in the scope above)